src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -188,46 +198,80 @@ private void EnqueueProcessSnapshot(Document newDocument)
                 }
             }
 
-            private async Task EnqueueProcessSnapshotWorkerAsync(Document document, CancellationToken cancellationToken)
+            private async Task EnqueueProcessSnapshotWorkerAsync(Document currentDocument, CancellationToken cancellationToken)
             {
                 // we will enqueue new one soon, cancel pending refresh right away
                 _reportChangeCancellationSource.Cancel();
 
-                var newText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
-                var snapshot = newText.FindCorrespondingEditorTextSnapshot();
-                if (snapshot == null)
+                var currentText = await currentDocument.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                var currentSnapshot = currentText.FindCorrespondingEditorTextSnapshot();
+                if (currentSnapshot == null)
                 {
                     // It's possible that we're seeing a notification for an update that happened
                     // just before the file was opened, and so the document we're given is still the
                     // old one.
                     return;
                 }
 
-                var latencyTracker = new RequestLatencyTracker(SyntacticLspLogger.RequestType.SyntacticTagger);
-                using (latencyTracker)
+                var service = TryGetClassificationService(currentSnapshot);
+                object currentCachedData = null;
+                var changedSpan = currentSnapshot.GetFullSpan();
+                if (service != null)
```

Comment:
```suggestion
                if (service is not null)
```

Reply:
FWIW, i do actually prefer the new form.  However, i think i'd like us to track this by investing in an analyzer/rule that applies this pattern consistently to our codebase.

Reply:
Agreed, analyzer would be nice